### ISSUE TITLE
Remove the `use-ssl` flag and logic

### ISSUE THREAD

**Issue by kenkendk** [duplicati/duplicati#5205]
The `use-ssl` feature was introduced over 15 years ago, when using SSL was costly and labor intensive. Especially the S3 protocol was implemented to run without SSL because of issues with certificates on various operating systems.

In the present, the situation is quite different, making TLS the norm and unencrypted traffic is considered legacy or debugging setups.

Unfortunately, simply removing the flag will cause backups to fail, so a transition period is needed.

Suggested process:

1. A new option `--disable-tls` should be implemented that has the inverse logic of `--use-ssl` (i.e., disabling ssl)
2. If both `use-ssl` and `disable-tls` are set, then prefer `disable-tls` and emit a warning.
3. If `use-ssl` is set, issue a deprecation warning and suggest the correct format. If it is `use-ssl=false` then suggest `disable-tls=true` and if it is true, suggest removing it.
4. Change the weird logic around appending an `s` to the backend protocol key (e.g., `s3 + use-ssl` = `s3s`), and make sure to give a warning if the `s` suffix is detected and suggest appropriate actions
5. Write an upgrade script that replaces `--use-ssl` and the `s` suffix in the settings database for users.

This will cover most users, but there may be some commandline instances who are relying on absence of `use-ssl` to mean no encryption.

We can either 
- allow these to break (giving a warning in the release notes) 
- or have a grace period where the absence of the `use-ssl` flag gives a warning.

In the second scenario we need to support the absence of the `disable-tls` for a while without it having any effect.

**Comment by ts678** [comment]
What's the GUI view of this? Right now, many backends have a "Use SSL" box which is unchecked by default.

One time I think I forgot to check it for an S3. IIRC it got rejected but I changed the credentials "just in case",
because it was easier than trying to figure out what S3 had revealed. If it was FTP, I think it's simple cleartext.

On that note, there are probably people with local on-LAN FTP or Minio servers who feel no need to do TLS.
Turning it off is easy though, and secure-by-default seems the safer plan to protect non-expert users better.

**Comment by kenkendk** [comment]
> What's the GUI view of this? Right now, many backends have a "Use SSL" box which is unchecked by default.
Yes, we should also opt-in for SSL via the UI. I have added it to the steps.

> One time I think I forgot to check it for an S3. IIRC it got rejected but I changed the credentials "just in case",
because it was easier than trying to figure out what S3 had revealed. If it was FTP, I think it's simple cleartext.

For S3 you can feel safe, that protocol was developed before SSL was common, so it signs each request in a special header and never sends your password to the backend. For FTP, yes, it will send the password in clear.

> On that note, there are probably people with local on-LAN FTP or Minio servers who feel no need to do TLS.
Yes, we need at least to support in-house setups where users do not observe the zero-trust setups. I think my upgrade path is considering this with the potential of requiring some command-line users needing to adjust their command to continue without encryption.

**Comment by Jojo-1000** [comment]
Maybe this should be decided per backend? For example FTPS is an established term to distinguish it from FTP, so I would suggest to keep the two URI versions. I think most people should be aware of the insecurity of plain FTP. In the UI, we can still enable the SSL option by default.

The others with the ssl option are S3, TAHOE and WebDav as far as I can tell. S3 and TAHOE should definitely be changed, I would even hide it in the advanced options. I'm not sure about WebDav, because it seems like that could often be locally hosted as well.

**Comment by kenkendk** [comment]
Good point about FTPS; I think we can register the backend with two protocol keys.

 I think "WEBDAVS" is not commonly used, but setting it to use TLS by default would make it more secure at the expense of those who may use it unencrypted in a LAN setting.

**Comment by luixxiul** [comment]
I experienced the connection reset issue on testing the connection while I was reviewing of https://github.com/duplicati/duplicati/pull/5345 as SSL was not enabled.

Since the console did not return the information about the error more than `System.IO.IOException: Unable to read data from the transport connection: Connection reset by peer`, at first I was not sure what caused the error.
